src/components/CouponDetails/ActionButton.jsx

```diff
@@ -0,0 +1,124 @@
+import { ACTION_TYPES, ACTION_LABELS, COUPON_FILTER_TYPES } from './constants';
+
+const ActionButton = ({
+  couponData: {
+    id,
+    errors,
+    available,
+    title,
+  },
+  selectedToggle,
+  code,
+  handleCodeActionSuccess,
+  setModalState,
+}) => {
+  const {
+    assigned_to: assignedTo,
+    is_public: isPublic,
+    redemptions,
+  } = code;
+  let remainingUses = redemptions.total - redemptions.used;
+
+  // Don't show `Assign/Remind/Revoke` buttons for an unavailable coupon
+  if (!available) {
+    return null;
+  }
+
+  // Don't show a button if all total redemptions have been used
+  if (redemptions.used === redemptions.total) {
+    return null;
+  }
+
+  const codeHasError = errors.find(errorItem => errorItem.code === code.code);
+  if (assignedTo) {
+    return (
+      <>
+        {!codeHasError && (
+          <>
+            <RemindButton
+              couponId={id}
+              couponTitle={title}
+              data={{
+                code: code.code,
+                email: code.assigned_to,
+              }}
+              onSuccess={response => handleCodeActionSuccess(ACTION_TYPES.remind, response)}
+            />
+            {' | '}
+          </>
+        )}
+        <RevokeButton
+          couponId={id}
+          couponTitle={title}
+          data={{
+            assigned_to: code.assigned_to,
+            code: code.code,
+          }}
+          onSuccess={response => handleCodeActionSuccess(ACTION_TYPES.revoke, response)}
+        />
+      </>
+    );
+  }
+
+  if (isPublic) {
+    return null;
+  }
+
+  // exclude existing assignments of code
+  if (selectedToggle === COUPON_FILTER_TYPES.unassigned) {
+    remainingUses -= redemptions.num_assignments;
+  }
+
+  return (
+    <Button
+      variant="link"
+      className="assignment-btn btn-sm p-0"
+      onClick={() => setModalState({
+        key: 'assignment',
+        options: {
+          couponId: id,
+          title,
+          data: {
+            code,
+            remainingUses,
+          },
+        },
+      })}
+    >
+      {ACTION_LABELS.assign}
+    </Button>
+  );
+};
+
+ActionButton.propTypes = {
+  couponData: PropTypes.shape({
+    id: PropTypes.number.isRequired,
+    errors: PropTypes.arrayOf(PropTypes.shape({
+      code: PropTypes.string,
+    })),
+    available: PropTypes.bool.isRequired,
+    title: PropTypes.string.isRequired,
+  }).isRequired,
+  selectedToggle: PropTypes.oneOf(Object.values(COUPON_FILTER_TYPES)).isRequired,
+  code: PropTypes.shape({
+    assigned_to: PropTypes.string,
+    is_public: PropTypes.bool.isRequired,
+    redemptions: PropTypes.shape({
+      total: PropTypes.number.isRequired,
+      used: PropTypes.number.isRequired,
+      num_assignments: PropTypes.number,
+    }),
+    code: PropTypes.string.isRequired,
+  }).isRequired,
+  handleCodeActionSuccess: PropTypes.func.isRequired,
+  setModalState: PropTypes.func.isRequired,
+};
+
+export default ActionButton;
```

Comment:
I may be losing my mind, but I don't see anywhere where we actually use ActionButton outside of the test files for it?
Everywhere else explicitly uses `ActionButtonWithModal` and that component *doesn't* call/inherit `ActionButton`
Is there some sort of implicit functionality I'm missing? (I even pulled the branch locally and searched through VSCode since it's more reliable than Github search and I'm still coming up with nothing outside of the base definition and test files.

Reply:
Let me check, it may have been lost in the rebase, it was a major rebase.

Reply:
Nope! It's there on line 411 of `src/components/CouponDetails/index.jsx`.